cliquet/storage/postgresql/client.py

```diff
@@ -17,41 +21,44 @@ def __init__(self, engine):
         #                                   loads=json.loads)
 
     @contextlib.contextmanager
-    def connect(self, readonly=False):
-        """Pulls a connection from the pool when context is entered and
+    def connect(self, readonly=False, force_commit=False):
+        """
+        Pulls a connection from the pool when context is entered and
         returns it when context is exited.
 
         A COMMIT is performed on the current transaction if everything went
         well. Otherwise transaction is ROLLBACK, and everything cleaned up.
-
-        XXX: Committing should not happen here but using a global transaction
-        manager like `pyramid-tm`.
         """
-        connection = None
-        trans = None
+        with_transaction = (not readonly and self.commit_manually)
```

Comment:
I know, it's a custom we have for boolean exprs. Do you want me to get rid of it ?